rudder-core/src/main/scala/com/normation/cfclerk/services/impl/GitTechniqueReader.scala

```diff
@@ -384,18 +384,29 @@ class GitTechniqueReader(
     }
   }
 
-  override def getTemplateContent[T](cf3PromisesFileTemplateId: Cf3PromisesFileTemplateId)(useIt : Option[InputStream] => T) : T = {
-    //build a treewalk with the path, given by Cf3PromisesFileTemplateId.toString
-    val path = cf3PromisesFileTemplateId.toString + Cf3PromisesFileTemplate.templateExtension
+  override def getTemplateContent[T](techniqueResourceId: TechniqueResourceId)(useIt : Option[InputStream] => T) : T = {
+    //build a treewalk with the path, given by TechniqueTemplateId
+    val filenameFilter = {
+      val name = techniqueResourceId.name + TechniqueTemplate.templateExtension
+      techniqueResourceId match {
+        case TechniqueResourceIdByName(tid, _) =>
+          new FileTreeFilter(canonizedRelativePath, s"${tid.name}/${tid.version.toString}/${name}")
+        case TechniqueResourceIdByPath(Nil, _) =>
```

Comment:
It will be on the PR for ticket http://www.rudder-project.org/redmine/issues/7402